lib/core/src/Cardano/Wallet/Address/Pool.hs

```diff
@@ -136,10 +138,21 @@ prop_fromIx Pool{addressFromIx,addresses} =
   where
     isGenerated addr (ix,_) = addressFromIx ix == addr
 
+prop_next_is_unused :: (Eq ix, Enum ix) => Pool addr ix -> Bool
```

Comment:
Hm. 🤔 I'm undecided whether this belongs in this module or in the `PoolSpec` unit tests:

* This property is not a property of the `Pool` data type itself, but it does clarify the relation between `nextIndex` and `Pool`. As such, I'm inclined to move it to the unit tests in PoolSpec.
* As a specification of the behavior of `nextIndex`, this property is not complete — however, the added comment for `nextIndex` would be complete, and one could specify it as an equation here. And once it's a complete specification, I'm inclined to keep it in this module.